### PR TITLE
[New] Add emergence.js w/ npm auto-update via single package.json

### LINKED ISSUE #12149: [Request] Add emergence.js
**Library name:** emergence.js
**Library description:** Detect element visibility in the browser
**Git repository url:** https://github.com/xtianmiller/emergence.js
**npm package name or url** (if there is one): https://www.npmjs.com/package/emergence.js
**License (List them all if it's multiple):** MIT
**Official homepage:** https://xtianmiller.github.io/emergence.js/

---
### ℹ️  introduction
CDNJS is one of the most famous free and public web front-end CDN services. We'd like to host a new web front-end library on CDNJS so that web developer can easily use this library on our free CDN without any additional download/upload process.

### 📚  How to add a library
There are two ways to send a pull request(PR):
- [ ] 1. [**Adding a new library via a single package.json**](https://github.com/cdnjs/cdnjs/blob/master/CONTRIBUTING.md#e-adding-a-new-library-by-a-single-packagejson)
  1. Creating a `package.json` under `ajax/libs/<LIBRARY_NAME>`
  2. Edit the `package.json`.
      - Required fields: `name`, `description`, `filename`, `license`, `repository`, `author`, `keywords`, auto-update config
      - Non-Required fields (If there is no field of the upstream): `homepage`
  3. Save the file and send a PR with the commit message as the followings example with npm auto-update (if you use git auto-update, please change from npm to git):
  This will be useful to us to maintain the git log and trace the history.
```
Add <LIBRARY_NAME> w/ npm auto-update via single package.json

close <#THIS_ISSUE_NUMBER>, cc <@AUTHOR>
```

- [ ] 2. [**Adding a new library with its assets**](https://github.com/cdnjs/cdnjs/blob/master/CONTRIBUTING.md#d-adding-a-new-library-with-its-assets)
  1. Creating a <LIBRARY_NAME> folder under `ajax/libs/`
  2. Edit the `package.json` under `ajax/libs/<LIBRARY_NAME>`.
      - Required fields: `name`, `description`, `filename`, `version`, `license`, `repository`, `author`, `keywords`, auto-update config
      - Non-Required fields (If there is no field of the upstream): `homepage`
  3. Create a folder named after the latest release version under `ajax/libs/<LIBRARY_NAME>`.
  4. Add the files from upstream to `ajax/libs/<LIBRARY_NAME>/<VERSION>`. The file with `md5sum` should as same as upstream.
  5. Save the file and send a PR with the commit message as the followings example with npm auto-update (if you use git auto-update, please change from npm to git):
  This will be useful to us to maintain the git log and trace the history.
```
Add <LIBRARY_NAME>@<VERSION> w/ npm auto-update

close <#THIS_ISSUE_NUMBER>, cc <@AUTHOR>
```

If you still don't know how to add a library, here is a tutorial in [document](https://github.com/cdnjs/cdnjs/blob/master/CONTRIBUTING.md), hope this will help you to understand.

### 💻  How to send a pull request
There are two ways to add a library, via GitHub website or locally.
- [ ] 1. **GitHub website**
- [ ] 2. **locally (Use Git command)** 
  - Using **sparseCheckout**
    Because cdnjs's repo is too huge, currently. It's better to use `sparseCheckout` to lets you only check out the files you want so that the whole directory can only occupy 2.2 GB in your computer. Here is a [tutorial](https://github.com/cdnjs/cdnjs/blob/master/documents/sparseCheckout.md) which let you know how to enable sparseCheckout.
  - If you don't know how to create a pull request locally, please take a look at this [tutotial](https://www.digitalocean.com/community/tutorials/how-to-create-a-pull-request-on-github).

### 🤖  How to setup auto-update config
The auto-update config can automatically add libraries to the latest version easily. [Here](https://github.com/cdnjs/cdnjs/blob/master/documents/autoupdate.md) is the document which can help you to know how to setup auto-update config. 
BTW, in `files` field, we'd like to let it be flexible but if there are some files not necessary to host, we will use some pattern to match and exclude them. There are many example configurations in [document](https://github.com/cdnjs/cdnjs/blob/master/documents/autoupdate.md), hope this will help you to understand. 

### :memo: Note
If you are interested in joinning us and help add this library, please feel free to check what way you'd like to **add a library** and  **send a pull request**. And let me know what kind of help you'll need, we'll help you send the valid pull request, thank you. 😃

### PR DESCRIPTION
**Pull request by ParkourKarthik**
close #12149, cc @xtianmiller

Pull request for issue: #12149 
Related issue(s): # #

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/xtianmiller/emergence.js
 * Official website (optional, not the repository): https://xtianmiller.github.io/emergence.js/
 * NPM package url (optional): https://www.npmjs.com/package/emergence.js
 * License and its reference: MIT
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 17
   - Count of stars: 721
   - Count of forks: 22
 * NPM download stats (optional):
   - Downloads in the last day: 27
   - Downloads in the last week: 234
   - Downloads in the last month: 379

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [ ] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 200 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [ ] More than 800 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [ ] Has valid tags for each versions (for git auto-update)
 * [ ] Auto-update setup
 * [ ] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [ ] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

**Comment by ParkourKarthik** [comment]
I expect suggestion for more keywords if possible.

**Comment by ParkourKarthik** [comment]
@PeterDaveHello  Does this build error have any changes to be done in the committed file?

**Comment by PeterDaveHello** [comment]
@ParkourKarthik hope, there was an issue with the CI server, fixed now.

File: ajax/libs/emergence.js/package.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  "description": "Detect element visibility in the browser",
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/xtianmiller/emergence.js.git"
+  },
+  "homepage": "https://xtianmiller.github.io/emergence.js/",
+  "keywords": [
+    "elements",
+    "emergence",
+    "DOM"
```
Comment (sufuf3) [inline ajax/libs/emergence.js/package.json]: Yes, it's good to add more keywords, such as https://github.com/xtianmiller/emergence.js/blob/master/bower.json#L11-L16
Maybe we can as the same with upstream. 😃 
What do you think?

Thank you.

**Comment by ParkourKarthik** [comment]
@sufuf3 Thank you :flushed:

File: ajax/libs/emergence.js/package.json (hunk 1 above)
Comment (ParkourKarthik) [reply ajax/libs/emergence.js/package.json]: Yeah! I think we could retain the same.

Comment (sufuf3) [reply ajax/libs/emergence.js/package.json]: @ParkourKarthik So would you modify this part to make this PR better to merge?
Thank you.

Comment (ParkourKarthik) [reply ajax/libs/emergence.js/package.json]: Sure :blush:

**Comment by sufuf3** [comment]
@cdnjs/active-contributor @cdnjs/intern3 @cdnjs/library-reviewer 
Please help review this PR, thanks.

**Comment by PeterDaveHello** [comment]
Well done @ParkourKarthik

**Pull request merged**